### ISSUE TITLE
In regards to the background of the model

### ISSUE THREAD

**Issue by MarKo9** [Oracen/MIDAS#4]
Hi,

First of all thanks for your work. 
I am in the process of doing some testings and if possible would like a quick clarification. Is your model the same or influenced by the one in the paper below? "MIDA: Multiple Imputation using Denoising Autoencoders" by Lovedeep Gondara and Ke Wang. 
https://arxiv.org/pdf/1705.02737.pdf

**Comment by Oracen-zz** [comment]
No, although I discovered that paper about 2 months after I started working on MIDAS. That concept has more in common with a chained equation approach (ie. MICE, Hmisc's areg.impute function) than MIDAS. In fact, it's more just like an ensemble of denoising autoencoders, meaning your training time scales with the number of imputations you require. MIDAS kind of draws on principles of variational inference to compute an approximate posterior. Train the model once, draw as many imputations as you require.

It's worth noting that Gondara and Wang don't exactly publicise their code or results. When I did eventually track down some code, it was on Gondara's github:
https://github.com/lgondara/loss-to-followup-DAE/blob/master/programs/Impute_LTF_DAE.py

If you look there, there's just an oblique Keras model, and it looks like noise isn't even dynamic for the representation they learn. Further, the loss function just appears to be MSE but the output layer is ReLU'd (ie. the output is constrained to be greater than zero). When combined with the StandardScaler transform they're using, they're going to have some difficulties. In the paper you discussed, they don't even address the issue of how missingness affects loss generation, which (from experience) is a key challenge. No mention of how they managed to bypass the challenge with softmax loss.

Also, it's worth noting; they changed the name of their paper in Feb 2018 by the looks of things. Their paper used to be called "Multiple Imputation Using Deep Denoising Autoencoders". Here's some correspondence with Andrew Gelman we had in Jan:
http://andrewgelman.com/2018/01/10/python-program-multivariate-missing-data-imputation-works-large-datasets/

In short; no we're not associated with them.

**Comment by Su-deep** [comment]
Hi,
While running the demo on jupyter notebook I ran in the following issue, while executing  the codeblock:
imputer.batch_generate_samples(m= 5)

print("Original value:", original_value)
imputed_vals = []
for dataset in imputer.output_list:
    imputed_vals.append(pd.DataFrame(scaler.inverse_transform(dataset),
                                     columns= dataset.columns).iloc[50, 0])
print("Imputed values:")
print(imputed_vals)
print("Imputation mean:", np.mean(imputed_vals))
print("Standard deviation of the imputation mean:", np.std(imputed_vals))

Can you please help me fix this. Thanks
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

---------------------------------------------------------------------------
INFO:tensorflow:Restoring parameters from tmp/MIDAS
Model restored.
---------------------------------------------------------------------------
InvalidIndexError                         Traceback (most recent call last)
<ipython-input-17-5baa6f0388d5> in <module>()
----> 1 imputer.batch_generate_samples(m= 5)
      2 
      3 print("Original value:", original_value)
      4 imputed_vals = []
      5 for dataset in imputer.output_list:

~/git/PycharmProjects/untitled/midas.py in batch_generate_samples(self, m, b_size, verbose)
    910                              columns= self.imputation_target.columns)
    911         output_df = self.imputation_target.copy()
--> 912         output_df[np.invert(self.na_matrix.values)] = y_out[np.invert(self.na_matrix.values)]
    913         self.output_list.append(output_df)
    914     return self

~/git/PycharmProjects/untitled/venv/lib/python3.5/site-packages/pandas/core/frame.py in __setitem__(self, key, value)
   3109 
   3110         if isinstance(key, DataFrame) or getattr(key, 'ndim', None) == 2:
-> 3111             self._setitem_frame(key, value)
   3112         elif isinstance(key, (Series, np.ndarray, list, Index)):
   3113             self._setitem_array(key, value)

~/git/PycharmProjects/untitled/venv/lib/python3.5/site-packages/pandas/core/frame.py in _setitem_frame(self, key, value)
   3158         self._check_inplace_setting(value)
   3159         self._check_setitem_copy()
-> 3160         self._where(-key, value, inplace=True)
   3161 
   3162     def _ensure_valid_index(self, value):

~/git/PycharmProjects/untitled/venv/lib/python3.5/site-packages/pandas/core/generic.py in _where(self, cond, other, inplace, axis, level, errors, try_cast)
   7543                         not all(other._get_axis(i).equals(ax)
   7544                                 for i, ax in enumerate(self.axes))):
-> 7545                     raise InvalidIndexError
   7546 
   7547             # slice me out of the other

InvalidIndexError:

**Comment by charuj** [comment]
Hi there @Oracen  and @ranjitlall ! 

Your repo is awesome! Really appreciate the work you put into it

Following-up with the above question, I'm wondering if you have a peer-reviewed (or preprint) paper about MIDAS? I'm wondering how I can cite your research to give you credit 

Thanks!

**Comment by Oracen-zz** [comment]
Hahaha good question!

My coauthor Ranjit and myself are currently authoring a paper on it. Ranjit
is a perfectionist and so it's taking a little longer than expected, but it
should be in preprint soon. Sorry I can't be more specific at this stage.

Regards,

Alex

**Comment by asthaIITD** [comment]
Hii @Oracen , 

Is the paper you both were authoring finalized? I could not find it online.

**Comment by tsrobinson** [comment]
Hi @asthallTD, @charuj,

We have now migrated MIDAS to [this new repo](https://github.com/ranjitlall/MIDAS) where all future releases will be made available.

The current paper explaining the background of the model can be found on APSA Preprints [here](https://doi.org/10.33774/apsa-2020-3tk40-v3): https://doi.org/10.33774/apsa-2020-3tk40-v3 

Best,
Tom
